code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/ProductCatalogPushTemplate.kt

```diff
@@ -65,56 +60,27 @@ internal class ProductCatalogPushTemplate : AEPPushTemplate {
         val uri: String
     )
 
-    constructor(data: Map<String, String>) : super(data) {
-        ctaButtonText = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.CATALOG_CTA_BUTTON_TEXT, null
-        )
-            ?: throw IllegalArgumentException("Required field \"${PushTemplateConstants.PushPayloadKeys.CATALOG_CTA_BUTTON_TEXT}\" not found.")
-        ctaButtonColor = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.CATALOG_CTA_BUTTON_COLOR, null
-        )
-            ?: throw IllegalArgumentException("Required field \"${PushTemplateConstants.PushPayloadKeys.CATALOG_CTA_BUTTON_COLOR}\" not found.")
-        ctaButtonUri = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.CATALOG_CTA_BUTTON_URI, null
-        )
-            ?: throw IllegalArgumentException("Required field \"${PushTemplateConstants.PushPayloadKeys.CATALOG_CTA_BUTTON_URI}\" not found.")
-        displayLayout = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.CATALOG_LAYOUT, null
-        )
-            ?: throw IllegalArgumentException("Required field \"${PushTemplateConstants.PushPayloadKeys.CATALOG_LAYOUT}\" not found.")
-        rawCatalogItems = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.CATALOG_ITEMS, null
-        )
-            ?: throw IllegalArgumentException("Required field \"${PushTemplateConstants.PushPayloadKeys.CATALOG_ITEMS}\" not found.")
+    /**
+     * Constructs a Product Catalog push template with the given NotificationData.
+     *
+     * @param data the data to initialize the push template with
+     * @param fromIntent flag to denote if the push template was built from an intent
+     * @throws IllegalArgumentException if any required fields for building the Product Catalog push template are missing
+     */
+    init {
+        ctaButtonText = data.getString(PushPayloadKeys.CATALOG_CTA_BUTTON_TEXT)
```

Comment:
prashanth had mentioned making a utility function to either retrieve the value from the map or throw an exception. could we use something like the following code within the MapData class to remove the need to throw the exceptions here:
```kotlin
 internal fun getNonNullOrThrow(
        data: Map<String, String>,
        key: String
    ): String {
        return DataReader.optString(data, key, null)
            ?: throw IllegalArgumentException("Required field \"$key\" not found.")
    }
```

Reply:
for intent we could add an extension function:
```kotlin
internal fun Intent.getNonNullOrThrow(key: String): String {
    return this.getStringExtra(key)
        ?: throw IllegalArgumentException("Required field \"$key\" not found.")
}
```

Reply:
Yeap, I did somthing similar. 
NotificaitonData will have method named "getRequiredString" that will do the operation like you mentioned
Will post the updated code soon